### DISCUSSION TITLE
How to fix search and sort in eloquent relationship?

### DISCUSSION THREAD

**Discussion by mahdijafaridev** [rappasoft/laravel-livewire-tables#561] [Q&A] [answered]
I want to be able to search and sort by eloquent relationships. I have a Transaction model that belongs to Partner. 
In table, I wrote this: 
 `Column::make('Partner', 'partners.name')->searchable()->sortable(),`

and when I search I get this error:
`BadMethodCallException
Call to undefined method App\Models\Transaction::partners()`

according to documentation: 

`To pull from a relationship, you can use the dot syntax:
Column::make('Permissions', 'abilities.permissions')`

but it doesn't work in my case.

**Comment by gal01azabolcs** [comment, marked as answer]
Hello, I think maybe this can help for you:
->searchable(function ($builder, $term) {
                    return $builder->orWhereHas('restaurant',function($query) use($term){
                        $query->where('name', 'LIKE', '%' . $term . '%');
                    });
                })
                ->sortable(function(Builder $query, $direction) {
                    return $query->orderBy(Restaurant::select('name')->whereColumn('products.restaurant_id', 'restaurants.id'), $direction);
                }),

**Comment by mahdijafaridev** [comment]
@gal01azabolcs thanks for your answer.
I tried this:
`Column::make('Partner', 'partner.name')->searchable()->sortable(),`. 
I removed `s` from `partners` and made it `partner.name` and now it searches partner name. But it does not sort and gives error. 

I think your answer will help me. thanks.

**Reply by gal01azabolcs** [reply]
@MahdiJafari1 with pleasure:D
